utils/tx.ts

```diff
@@ -98,14 +101,30 @@ export async function calculate_sell_price_id_rpc(soldTokenId: BN, boughtTokenId
 }
 
 export async function getCurrentNonce(account?: string) {
-  const api = getApi();
-  if (account) {
-    const { nonce } = await api.query.system.account(account);
-    return new BN(nonce.toString())
+
+  const {sudo:sudoUserName} = getEnvironmentRequiredVars();
+  const sudo = new User(new Keyring({ type: 'sr25519' }), sudoUserName);
+  // lets check if sudo -> calculate manually nonce.
+  if(account === sudo.keyRingPair.address){
+      const nonce = new BN(await SudoDB.getInstance().getSudoNonce(account));
+
+      return nonce;
+
+  }else if (account){
+
+    return getChainNonce(account);
+
   }
   return new BN(-1)
```

Comment:
i know that its not part of this PR but shouldnt some error be reported instead

Reply:
AFIK ( an I am not 100% sure ) a nonce of -1 is a "default" value for nonce. 
But! if you are sure that is not valid, I'll trigger an exception instead, that would make more sense.